### DISCUSSION TITLE
Questions about the client server

### DISCUSSION THREAD

**Discussion by MacTracy** [openTCS/opentcs#58] [Q&A]
At present, openTCS is the client and the vehicle is the server; Am I right?
Can openTCS be used as the server and the vehicle as the client?

**Comment by swltr** [comment]
You can implement any kind of communication between openTCS and a vehicle by using a fitting vehicle driver. See [the Developer's Guide](https://www.opentcs.org/docs/5.6/developer/developers-guide/opentcs-developers-guide.html#_developing_vehicle_drivers) for how to create your own vehicle driver.
